### PR TITLE
feat(docker): Expand Dockerfile with SCIP-Python and indexing scripts #398

### PR DESCRIPTION
**Pull request by Exarchias**
This PR aims to address issue #398 which involves enhancing our Dockerfile to provide more functionality for our development workflow.

The issue specifies the need to include SCIP-Python and various indexing scripts as part of the Docker container's setup. These additions are important for code analysis and development workflow efficiency.

Therefore, the proposal is to modify the Dockerfile to include these components and scripts during the Docker build process. 

However, this is not a straightforward task as it requires a method to handle the automata configure command, which sets up the environment and configuration files. The objective is to find a solution that allows these configurations to be set gracefully during the build process.

This PR represents the initiative to devise such a solution and to enhance the efficiency and effectiveness of our development and testing environments by extending the Docker container's capabilities.

Please note that this is a preliminary solution and it is still a work in progress. It has not been fully tested and does not yet comply with Codacy standards. Therefore, the issue is not yet fully resolved with this PR. Further contributions and discussions are welcome to refine and finalize the solution.

### CODE REVIEW

**Comment by NolanTrem** [comment]
LGTM! Can we just update the readme Installation via Docker section to show the updated run command?

File: Dockerfile [hunk 1]
```diff
@@ -17,11 +17,24 @@ RUN poetry install
...
+
 # Instructions on how to run the Docker container
 # These instructions will not be executed during the Docker build
 # They are merely here for the user's convenience
 # Run these commands in the terminal to build and run the Docker container:
-# docker build -t automata .
-# docker run -it --name test automata bash
-# docker stop test
-# docker rm test
+# docker run --name automata_container -it --rm -e OPENAI_API_KEY=<your_openai_key> -e GITHUB_API_KEY=<your_github_key> ghcr.io/emrgnt-cmplxty/automata:latest
```
Comment (NolanTrem) [inline Dockerfile:38]: This is the command that will need to be updated in the Readme

Comment (Exarchias) [reply Dockerfile:38]: Totally agree! I will do it during this PR :)

Comment (Exarchias) [reply Dockerfile:38]: And it is done! Good call @NolanTrem!

**Review by NolanTrem — APPROVED** [review]
LGTM!

**Comment by Exarchias** [comment]
Thank you for your help, @NolanTrem!

**Pull request merged**